go.mod

```diff
@@ -1,35 +1,69 @@
 module github.com/keikoproj/addon-manager
 
-go 1.16
+go 1.17
 
 require (
-	cloud.google.com/go v0.65.0 // indirect
 	github.com/Masterminds/semver/v3 v3.1.1
 	github.com/go-logr/logr v0.4.0
-	github.com/go-logr/zapr v0.2.0 // indirect
-	github.com/google/go-cmp v0.5.2 // indirect
-	github.com/google/gofuzz v1.2.0 // indirect
-	github.com/google/uuid v1.1.2 // indirect
-	github.com/googleapis/gnostic v0.5.1 // indirect
-	github.com/imdario/mergo v0.3.11 // indirect
 	github.com/jinzhu/inflection v1.0.0
 	github.com/onsi/ginkgo v1.16.2
 	github.com/onsi/gomega v1.11.0
 	github.com/pkg/errors v0.9.1
-	github.com/prometheus/common v0.13.0 // indirect
 	github.com/spf13/cobra v1.0.0
-	go.uber.org/zap v1.15.0 // indirect
-	golang.org/x/crypto v0.0.0-20200820211705-5c72a883971a // indirect
 	golang.org/x/net v0.0.0-20201202161906-c7110b5ffcbb
-	golang.org/x/time v0.0.0-20200630173020-3af7569d3a1e // indirect
-	gomodules.xyz/jsonpatch/v2 v2.1.0 // indirect
 	gopkg.in/yaml.v3 v3.0.0-20210107192922-496545a6307b
 	k8s.io/api v0.19.11
 	k8s.io/apiextensions-apiserver v0.19.11
```

Comment:
When I said k8s API updated it's these dependencies, they should be upgraded to 1.21 at least.
If we want to go to 1.22 it's possible but we need to verify compatibility first.

I don't see you've updated these which you can file a new issue for these.

Reply:
Right. My original plan is to include the "deprecating old version" in this PR, working on it. 
Thus, revert it WIP.

Reply:
92347195c25c4e212f79db24b0a93633ed0675bb